app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", only: :destroy 
+  
+  def create
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.create(comments_params)
+
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.find(params[:id])
+    @comment.destroy
+
+    redirect_to(article_path(@article))
+  end
+
+  private 
```

Comment:
Remember to remove trailing spaces. You can configure your editor to do it automatically.

Reply:
In this case the code will be written like this ?

   ```
      redirect_to(article_path(@article))
end
private 
```

Reply:
Nope, trailing spaces are empty spaces at the end of the lines.

The empty line at `line:19` is ok.